docs/python-grader/sphinx-docs.md

```diff
@@ -62,6 +62,30 @@ Check that a student list has correct length with respect to a reference list. C
 
 - `report_failure`: If true, feedback will be given on failure.
 
+#### classmethod check_dict(name, ref, data)
+
+Checks that a student dict has all correct key-value mappings with respect to a reference dict. Can also check that a student dict has correct length of keys with respect to a reference dict's length of keys.  Can also check for a homogeneous data type for either the dict's key, dict's value, or both. Can also check if only some particular keys present in the student dict or not. Can also check student dict with respect to a reference dict only based on either keys, values, or both.
```

Comment:
I think this is a little confusing. The `check_only_keys` works mostly the way I would expect, but `check_only_values` checks that the set of values is equal, but ignores the keys they are mapped to. I think we should get rid of the `check_only_values` flag but keep the `check_only_values` flag (defaulting to full equality if this flag is false).

Reply:
I believe the current code covers this functionality.  

i.e. 
if `check_only_values` flag is True, then we solely evaluate the student's submission based on the "set of values is equal, but ignores the keys they are mapped to." 

and if `check_only_values` flag is False, then we perform the proper full equality i.e. `ref == data` Line 391.

Maybe I need to write the documentation more elaborately. 

@eliotwrobson Is this what you are looking for? or please let me know if you need any chances here.

Reply:
I think that makes sense. More detailed documentation would make the logic much easier to follow.

Reply:
Awesome! I will update the documentation to keep it unambiguous